### DISCUSSION TITLE
Acentuação com engine Indy

### DISCUSSION THREAD

**Discussion by cauenishijima** [viniciussanchez/RESTRequest4Delphi#226] [Ideas]
Olá a todos,

Estou utilizando o RR4D com a engine Indy, pois achei o desempenho superior em relação à engine default. No entanto, ao fazer essa mudança, comecei a enfrentar problemas com a acentuação, que passou a ser exibida incorretamente. Esse problema não ocorria quando eu usava a engine default.

Ao analisar o código, fiz a seguinte alteração na unit **RESTRequest4D.Response.Indy**:

```
function TResponseIndy.Content: string;
var
  LContent: string;
begin
  LContent := TStringStream(FIdHTTP.Response.ContentStream).DataString;
  Result   := TEncoding.UTF8.GetString(TEncoding.ANSI.GetBytes(LContent));
  
  // Código original
  // Result := TStringStream(FIdHTTP.Response.ContentStream).DataString;
end;
```

Não sei se essa modificação pode causar algum impacto ou se é aplicável a outras engines, mas essa foi a solução que usei para corrigir o problema. Espero que ajude quem estiver passando pela mesma situação.

**Comment by fabianosafe** [comment]
Estava com o mesmo problema, inclusive, quando utiliza o .Adapters(
fiz uma alteração no RESTRequest4D.Request.Indy resolvendo isso

em constructor TRequestIndy.Create;

//  FStreamResult := TStringStream.Create;
  FStreamResult := TStringStream.Create('', TEncoding.UTF8);

**Reply by cauenishijima** [reply]
Opa @fabianosafe beleza?

Acho que sua solução é até melhor e mais simples, testei aqui e funcionou certinho, valeu 🚀

**Reply by viniciussanchez** [reply]
Manda um pull request lá ajustando
